contracts/evmscript/executors/CallsScript.sol

```diff
@@ -50,11 +56,43 @@ contract CallsScript is BaseEVMScriptExecutor {
 
             bool success;
             assembly {
-                success := call(sub(gas, 5000), contractAddress, 0, calldataStart, calldataLength, 0, 0)
-            }
+                success := call(
+                    sub(gas, 5000),       // forward gas left - 5000
+                    contractAddress,      // address
+                    0,                    // no value
+                    calldataStart,        // calldata start
+                    calldataLength,       // calldata length
+                    0,                    // don't write output
+                    0                     // don't write output
+                )
 
-            require(success, ERROR_CALL_REVERTED);
+                switch success
+                case 0 {
+                    let ptr := mload(0x40)
+
+                    switch returndatasize
+                    case 0 {
+                        // No error data was returned, revert with "EVMCALLS_CALL_REVERTED"
+                        // See remix: doing a `revert("EVMCALLS_CALL_REVERTED")` always results in
+                        // this memory layout
+                        mstore(ptr, 0x08c379a000000000000000000000000000000000000000000000000000000000)
+                        mstore(add(ptr, 0x20), 0x0000002000000000000000000000000000000000000000000000000000000000)
+                        mstore(add(ptr, 0x40), 0x0000001645564d43414c4c535f43414c4c5f5245564552544544000000000000)
+                        mstore(add(ptr, 0x60), 0)
+
+                        revert(ptr, 100)
```

Comment:
It also may be clearer using correctly padded data with fixed indexes, it is more readable I think, and it somehow mimics what the EVM actually does.
```sol
mstore(ptr, 0x08c379a000000000000000000000000000000000000000000000000000000000)         // error identifier
mstore(add(ptr, 0x04), 0x0000000000000000000000000000000000000000000000000000000000000020) // starting offset
mstore(add(ptr, 0x24), 0x0000000000000000000000000000000000000000000000000000000000000016) // reason length
mstore(add(ptr, 0x44), 0x45564d43414c4c535f43414c4c5f524556455254454400000000000000000000) // reason
```